src/cmd/flux-account-update-db.py

```diff
@@ -212,40 +213,41 @@ def update_db(path, new_db):
     old_cur = old_conn.cursor()
 
     try:
-        # we should only pass a new DB if we are testing the flux
-        # account-update-db command; normally, no value should be passed for
-        # new_db
-        if not new_db:
-            new_db = "tmpFluxAccounting.db"
-            c.create_db(new_db)
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            # we should only pass a new DB if we are testing the flux
+            # account-update-db command; normally, no value should be passed for
+            # new_db
+            if not new_db:
+                tmp_db_path = tmp_dir_name + "/tmpFluxAccounting.db"
+                c.create_db(tmp_db_path)
 
-        new_conn = sqlite3.connect("file:%s?mode=rw" % new_db, uri=True)
-        new_cur = new_conn.cursor()
+                new_conn = sqlite3.connect("file:%s?mode=rw" % (tmp_db_path), uri=True)
+            else:
+                new_conn = sqlite3.connect("file:%s?mode=rw" % new_db, uri=True)
 
-        update_tables(old_conn, old_cur, new_cur)
+            new_cur = new_conn.cursor()
 
-        update_columns(old_conn, old_cur, new_cur)
+            update_tables(old_conn, old_cur, new_cur)
 
-        # update user_version for DB
-        old_cur.execute(
-            "PRAGMA user_version = %d" % (fluxacct.accounting.db_schema_version)
-        )
+            update_columns(old_conn, old_cur, new_cur)
 
-        # commit changes
-        old_conn.commit()
+            # update user_version for DB
+            old_cur.execute(
+                "PRAGMA user_version = %d" % (fluxacct.accounting.db_schema_version)
+            )
 
-        # close connections to DB's and remove temporary database
-        old_conn.close()
-        new_conn.close()
+            # commit changes
+            old_conn.commit()
 
-        os.remove(new_db)
+            # close connections to DB's and remove temporary database
+            old_conn.close()
+            new_conn.close()
     except sqlite3.OperationalError as exc:
         print(f"Unable to open temporary database file: {new_db}")
         print(f"Exception: {exc}")
         sys.exit(1)
     except sqlite3.IntegrityError as exc:
         print(f"Exception: {exc}")
-        os.remove(new_db)
```

Comment:
The temporary directory should also be removed, not just the db file, and preferably on all exit paths.

Reply:
Oh, yes, thank you for pointing this out. I just force-pushed a fix to call to remove the directory (and its files, which should just be the temporary database) on a Python Exception. When `update-db` succeeds, the directory and files are cleaned up automatically upon completion (I'm looking at [this](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) explanation on the Python docs).